src/test/java/uk/gov/pay/directdebit/events/dao/SandboxEventDaoIT.java

```diff
@@ -59,4 +63,38 @@ public void shouldInsertAnEvent() throws IOException {
         assertThat(foundSandboxEvent.get("event_cause"), is("anEventCause"));
         assertThat((Timestamp) foundSandboxEvent.get("created_at"), isDate(CREATED_AT));
     }
+
+    @Test
+    public void shouldFindLatestApplicableEventForPayment() {
```

Comment:
I like the various test fixtures you set up to help assure the logic is right, seems reasonable and complete since we search on `paymentId`, `event_action` and `created_at` 👍